anta/tests/connectivity.py

```diff
@@ -89,19 +90,42 @@ def render(self, template: AntaTemplate) -> list[AntaCommand]:
             for host in self.inputs.hosts
         ]
 
+    def _is_host_reachable(self, host: Host, message: str) -> None:
+        """Check if a host is reachable."""
+        # Verifies the network is reachable
+        if host.reachable and f"{host.repeat} received" not in message:
+            self.result.is_failure(f"{host} - Unreachable")
+
+        # Verifies the network is unreachable
+        if not host.reachable and f"{host.repeat} received" in message:
+            self.result.is_failure(f"{host} - Destination is expected to be unreachable but found reachable")
+
     @AntaTest.anta_test
     def test(self) -> None:
         """Main test function for VerifyReachability."""
         self.result.is_success()
 
         for command, host in zip(self.instance_commands, self.inputs.hosts):
-            # Verifies the network is reachable
-            if host.reachable and f"{host.repeat} received" not in command.json_output["messages"][0]:
-                self.result.is_failure(f"{host} - Unreachable")
+            message = command.json_output["messages"][0]
+
+            # Extract the command failure details
+            repeat_count = "|".join(str(i) for i in range(host.repeat + 1))
+            pattern = re.compile(rf"(?:{repeat_count})\s+received")
```

Comment:
Just to make it clear that we are building a pattern.
```suggestion
            repeat_count_pattern = "|".join(str(i) for i in range(host.repeat + 1))
            pattern = re.compile(rf"(?:{repeat_count_pattern})\s+received")
```